src/transformStateWithClones.js

```diff
@@ -8,6 +8,29 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let newState = { ...state };
+  const result = [];
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(newState, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key2 of obj.keysToRemove) {
+          delete newState[key2];
+        }
+        break;
+
+      default:
+        newState = {};
```

Comment:
I think it isn`t correct write this operation in default. It will be better if you use one more case with 'clear', use loop (key in obj) and add empty object into copied object inside this loop.